lighthouse-core/lib/sentry.js

```diff
@@ -74,6 +76,19 @@ function init(opts) {
       // @ts-ignore Non-standard property added to flag error as not needing capturing.
       if (err.expected) return;
 
+      const tags = opts.tags || {};
+      if (tags.audit) {
+        const key = `audit-${tags.audit}-${err.message}`;
+        if (sentryExceptionCache.has(key)) return;
+        sentryExceptionCache.set(key, true);
+      }
+
+      if (tags.gatherer) {
+        const key = `gatherer-${tags.gatherer}-${err.message}`;
+        if (sentryExceptionCache.has(key)) return;
```

Comment:
Why check `has`?  Can't it just `set` it and override the old version in the map?  Or is that not readable JS? :confused:

Reply:
if it has the key we want to `return` and skip the rest of execution, not just the rest of the block. worth adding a comment since it's confusing already? :)